components/purchase_modal/process_payment_setup.tsx

```diff
@@ -0,0 +1,169 @@
+
+        this.state = {
+            progress: 0,
+            error: '',
+            state: ProcessState.PROCESSING,
+        };
+    }
+
+    public componentDidMount() {
+        this.savePaymentMethod();
+
+        this.intervalId = setInterval(this.updateProgress, MIN_PROCESSING_MILLISECONDS / MAX_FAKE_PROGRESS);
+    }
+
+    public componentWillUnmount() {
+        clearInterval(this.intervalId);
+    }
+
+    private updateProgress = () => {
+        let {progress} = this.state;
+
+        if (progress >= MAX_FAKE_PROGRESS) {
+            clearInterval(this.intervalId);
+            return;
+        }
+
+        progress += 1;
+        this.setState({progress: progress > MAX_FAKE_PROGRESS ? MAX_FAKE_PROGRESS : progress});
+    }
+
+    private savePaymentMethod = async () => {
+        const start = new Date();
+        const {stripe, addPaymentMethod, billingDetails, isDevMode} = this.props;
+        const errorText = await addPaymentMethod((await stripe)!, billingDetails!, isDevMode);
+
+        if (errorText) {
+            this.setState({
+                error: errorText,
+                state: ProcessState.FAILED});
+            return;
+        }
+
+        const end = new Date();
+        const millisecondsElapsed = end.valueOf() - start.valueOf();
+        if (millisecondsElapsed < MIN_PROCESSING_MILLISECONDS) {
+            setTimeout(this.completePayment, MIN_PROCESSING_MILLISECONDS - millisecondsElapsed);
+            return;
+        }
+
+        this.completePayment();
+    }
+
+    private completePayment = () => {
+        clearInterval(this.intervalId);
+        this.setState({state: ProcessState.SUCCESS, progress: 100});
+    }
+
+    private handleGoBack = () => {
+        clearInterval(this.intervalId);
+        this.setState({
+            progress: 0,
+            error: '',
+            state: ProcessState.PROCESSING,
+        });
+        this.props.onBack();
+    }
+
+    public render() {
+        const {state, progress} = this.state;
+
+        const progressBar: JSX.Element | null = (
+            <div className='ProcessPayment-progress'>
+                <div
+                    className='ProcessPayment-progress-fill'
+                    style={{width: `${progress}%`}}
+                />
+            </div>
+        );
+
+        switch (state) {
+        case ProcessState.PROCESSING:
+            return (
+                <IconMessage
+                    title={t('admin.billing.subscription.verifyPaymentInformation')}
+                    subtitle={''}
+                    icon={processSvg}
+                    footer={progressBar}
+                />
+            );
+        case ProcessState.SUCCESS:
+            return (
+                <IconMessage
+                    title={t('admin.billing.subscription.upgradedSuccess')}
+                    subtitle={
+                        'Starting August 8, 2020 you will be charged based on the number of enabled users'
```

Comment:
We need to change this date to use the first date of the next month as we are doing in the modal here https://github.com/mattermost/mattermost-webapp/blob/46247a6f71a288694af34dfa9348d7c2f506536f/components/purchase_modal/purchase_modal.tsx#L83